juniper_official/Kernel/check-too-many-dead-ifstates.rule

```diff
@@ -0,0 +1,94 @@
+/*
+ * This rule checks for the number of dead ifstates at any given time in
+ * the system. A dead Ifstate gets garbage collected once all the consumers
+ * see the corresponding delete message. If these dead ifstates continue to
+ * pile up, this indicates a garbage collection problem and can eat up
+ * significant resources such as memory and indices.
+ *
+ * This rule monitors the number of dead Ifstates and raises an alarm if there
+ * are too many of these (> 50% of alive ifstates) detected for significant
+ * duration of time (1 hour).
+ */
+iceberg {
+    topic kernel.ifstate {
+        rule check-too-many-dead-ifstates {
+            function alive-dead-ifs-ratio {
+                description "This returns the number of alive/dead ifstates count ratio";
+                path get-dead-alive-ratio.py;
+                method get_dead_alive_ratio;
+                argument alive_ifstates_cnt {
+                    mandatory;
+                }
+                argument dead_ifstates_cnt {
+                    mandatory;
+                }
+            }
+            sensor kernel-ifstate-sensor {
+                open-config {
+                    sensor-name /junos/kernel-ifstate;
+                    frequency 10m;
+                }
+            }
+            field alive-ifstates-cnt {
+                sensor kernel-ifstate-sensor {
+                    path /junos/kernel-ifstate/alive-ifstates-cnt;
+                }
+                type integer;
+                description "Alive ifstates counter";
+            }
+            field dead-ifstates-cnt {
+                sensor kernel-ifstate-sensor {
+                    path /junos/kernel-ifstate/dead-ifstates-cnt;
+                }
+                type integer;
+                description "Dead ifstates counter - OC sensor";
+            }
+            field ratio-value {
+                formula {
+                    user-defined-function {
+                        function-name alive-dead-ifs-ratio;
+                        argument dead_ifstates_cnt "$dead-ifstates-cnt";
+                        argument alive_ifstates_cnt "$alive-ifstates-cnt";
+                    }
+                }
+                type integer;
+                description "This field has the value of final decision if alive/dead ifstates in the system are normal";
+            }
+            trigger report-dead-alive-ifstates-ratio {
```

Comment:
1) Provide trigger description and synopsis detailing sensor info.
2) Trigger frequency is must as default frequency is 10s and trigger gets evaluated once in 10s. but here sensor frequecy is 10 mins hence trigger frequency can be greaterthan equal to 10 mins.
e.g. synopsis "Routing engine GRES kpi";
       description "sets health based on Routing engine CPU utilization for 15 minutes";

Reply:
Addressed.